lms/djangoapps/courseware/grades.py

```diff
@@ -127,6 +127,51 @@ def get(self, location):
         return max_score
 
 
+class ProgressSummary(object):
+    """
+    Wrapper class for the computation of a user's scores across a course.
+
+    Attributes
+       chapters: a summary of all sections with problems in the course. It is
+       organized as an array of chapters, each containing an array of sections,
+       each containing an array of scores. This contains information for graded
+       and ungraded problems, and is good for displaying a course summary with
+       due dates, etc.
+
+       weighted_scores: a dictionary mapping module locations to weighted Score
+       objects.
+
+       locations_to_children: a dictionary mapping module locations to their
+       direct descendants.
+    """
+    def __init__(self, chapters, weighted_scores, locations_to_children):
+        self.chapters = chapters
+        self.weighted_scores = weighted_scores
+        self.locations_to_children = locations_to_children
+
+    def score_for_module(self, location):
+        """
+        Calculate the aggregate weighted score for any location in the course.
+        This method returns a tuple containing (earned_score, possible_score).
+
+        If the location is of 'problem' type, this method will return the
+        possible and earned scores for that problem. If the location refers to a
+        composite module (a vertical, section or chapter) the scores will be the
```

Comment:
You can't pull the score for anything higher than a section without getting into complicated rules. A section (sequence) can be computed in the manner below, but a chapter might have sections that are of different types (HW vs. Exam) and that involves grading policy weights, drop rules, and a bunch of stuff you probably don't want to worry about. For now, please restrict this class to talking about assignments at that lower level.